modules/dists/CyclicDist.chpl

```diff
@@ -585,56 +585,55 @@ iter CyclicDom.these(param tag: iterKind) where tag == iterKind.leader {
   const ignoreRunning = dist.dataParIgnoreRunningTasks;
   const minSize = dist.dataParMinGranularity;
   const wholeLow = whole.low;
+  const wholeStride = whole.stride;
+
```

Comment:
I feel pretty confident you don't need to play games with ignoringRunning anymore (but I'm never certain when it comes to the running task count.)

For this case `here.runingTasks()` should always be 1 on the initiating locale after the `coforall locDom in locDoms do on locDom` (and there shouldn't be any possibility for a race.) 

That coforall turns into something like:

```chpl
runningTaskCount.sub(1);
endCount.add(locDoms.size);

for locDom in locDoms {
  executeOnNB(locDom.loc, wrapperFunc, ...);
}

endCount.waitFor(0);
runningTaskCount.add(1);

proc wraperFunc() {
  runningTaskCount.add(1);
  realFunc();
  runningTaskCount.sub(1);
  endCount.sub(1);
}
```

(We end up incrementing/decrementing runningTaskCount on the initiating locale more than we strictly need to, but that helps keep the implementation somewhat sane.)

Reply:
I was only trying to bring CyclicDist in line with what BlockDist does.  I assume what you're saying applies to BlockDist as well?  If so, I don't think I want to take that on in this PR since it's not directly related to what I'm trying to do (reduce comm counts).

Reply:
Oh, for some reason I thought we removed this workaround in block, but you're right it's still there.

Reply:
I've opened https://github.com/chapel-lang/chapel/issues/13025 to capture that